meetings/2021-09-22.md

```diff
@@ -0,0 +1,159 @@
+* **Kjetil**: Can we agree that Read and Write can be top level classes. We could reach consensus already.
+* **Sarven**: yes. they're class of operations. is there no consensus on that?
+* **Kjetil**: Justin mentioned Write could be deprecated.. but that's not the case for a lot of social groups. Idea of having RW as upper classes .. can we have consensus on that.
+* **Henry**: yes.. +1 for Read and Write as top level operations. That fits perfectly  with lenses. It is mind-boggling to think that this simple concept is behind so much (OO programming, Dyanamic systems, ...)
+* **Sarven**: yes.
+* **Pavlik**: WAC's RW or ?
+* **Henry**: generic.
+* **Sarven**: RW *operations*
+* **Henry**: Rread is observation with no change of state, Write is transformation of state.
+* .. :Control is not an operation in the way those  two key elements are. Control is of course an essential part of "access control", as per the work from 1992 [Access Control in distributed systems](https://dl.acm.org/doi/abs/10.1145/155183.155225). What Abadi, Burrows, Lambson and Plotkin started developing there is that control is built on the notion of "saying that". In particular in N3 this gives us
+```Turtle
+{ ?s controls ?p } => { ?s says ?p. ?p a log:Truth }
+```
+i.e. someone controls something if when they says something about it, it is true. E.g. a home owner can says don't go into house, and thereby it is true. He has the last decision. This is what is known as a  speech act. It is about making true by saying, rather than observing that something is true. 
+* **Sarven**: I don't think there are big differences between ACL and ACP. It's not like capability vs ACL.
+* **Pavlik**: 
+* **Kjetil**: we're also talking about modify, create, delete.. and so on. how many operations will we bring in to the specification. Also we have to consider how the HTTP protocol sees them. There are a lot of things there to consider.
+* **Pavlik**: I find it confusing to call both operation and access mode 'Read', 'Write'.
+* **Henry**: Read is an observation function on a state S to some representation A
+```scala
+   read: S -> A
+```
+Write is a function that gives a representation A and a state and changes it to a new state
+```scala
+   write: S × A -> S 
+  ```
+* **Sarven**: ...
+* **Sarven**: We can introduce new access model to Delete, it seems to just substitute variable name but end result seems the same.
+* **Henry**: I think Justin should try to write how this example would work with WAC, and show what Delete would give us.
+* **Sarven**: too much intelectualising not enough visible implementations.
+* **Henry**: I will write client libraries next month, so I can try out these ideas.
+* **Pavlik**: re Read operation on the resource.. ACL seems the nuance of e.g. Read mode.
+* **Matthieu**: wondering about UCs and think Henry is right.. Write on. !!something missing here 
+* **Sarven**: we have issue about error codes
```

Comment:
@matthieubosquet there is something missing here. Not sure what it was.

Reply:
Suggested something that I think sumarizes what I thought at the time.